### ISSUE TITLE
No option to select node template when creating RKE2 cluster pool

### ISSUE THREAD

**Issue by johnwc** [rancher/dashboard#6020]
**Setup**
- Rancher version: 2.6.4
- Browser type & version: Edge 101.0.1210.39

**Describe the bug**
When creating or editing a cluster, their is no option to choose a node template when setting up pools.

Creating this issue per request from @rak-phillip in response to comment https://github.com/rancher/dashboard/issues/3553#issuecomment-1134837914

**To Reproduce**
Create a new cluster, configure pool.

**Result**
No option to select node template.

**Expected Result**
Able to select a node template that was previously created.

**Screenshots**
<img width="1213" alt="image" src="https://user-images.githubusercontent.com/2798441/169859407-1fcadad7-1606-4031-b600-1bffe94f600e.png">

**Additional context**
We are deploying using vSphere if that makes a difference.

**Context**
With it being unable to use a template, that we can update at a later time we have ran into the scenario of...
We updated our VM template in vSphere, which caused the `(3)` that was suffixed at the end of the template name to be incremented to `(4)`. Now we can't increase our node count because it's pointing to a non-existing template name. I came to the cluster config to edit it to the new template name, but it will not update at all. How do I update the template in the cluster config? We're not expecting it to redeploy nodes with the new template, just use the new template on new nodes.

**Comment by richard-cox** [comment]
I'm a vSphere noob, but I think there's cross-over with https://github.com/rancher/dashboard/issues/5981. We should evaluate both issues at the same time

**Comment by johnwc** [comment]
Yes, that seems to be what we are after. Also please note the scenario described at the end of the issue above, of needing to update the cluster's pool's vSphere template after a cluster is created.

**Comment by catherineluse** [comment]
> How do I update the template in the cluster config? We're not expecting it to redeploy nodes with the new template, just use the new template on new nodes.

Because machine templates are immutable in CAPI, https://cluster-api.sigs.k8s.io/tasks/updating-machine-templates.html, it is not possible to update the existing machine template. The only way to change the machine configuration is to create a new machine template.

I think in this case, the best solution would be to create a new node pool that refers to the incremented template in VSphere.

**Comment by johnwc** [comment]
I think you are confusing a machine template with what actually uses the machine template, like a machine set. It states directly in the link you provided that you can change a machine set to use a different machine config. In my case I'd need to create a new machine config using the new vsphere template name, then edit the machineset with the new machine config name. This is the part that is missing in rancher.

**Comment by catherineluse** [comment]
I don't think we can change the MachineSets directly. As I understand it, MachineSets aren't meant to be edited by humans. They are analogous to ReplicaSets in that they are used to keep track of how many items in are in each group as an update is getting rolled out, except instead of Pods they are tracking machines.

**Comment by johnwc** [comment]
I states directly in the link you provided earlier on how to edit the MachineSet to point to a new template. Is a pool not a MachineSet?

> For a MachineDeployment or MachineSet, users would need to modify the spec.template.spec.infrastructureRef.name field. In all cases, the name field should be updated to point to the newly-modified infrastructure machine template.

**Comment by catherineluse** [comment]
Yes, a MachineDeployment and MachineSet are different resources in CAPI and the docs are contradicting themselves, because on that page they mention editing a MachineSet, but on another page they say you should only edit the MachineDeployment because the MachineSets are managed by Kubernetes. Next week I'll open a documentation issue in the CAPI docs repo to that effect.
